src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -73,6 +73,15 @@ static bool is_vector_mask(ciKlass* klass) {
   return klass->is_subclass_of(ciEnv::current()->vector_VectorMask_klass());
 }
 
+static bool is_maskall_type(const TypeLong* type, int vlen) {
+  if (!type->is_con()) {
+    return false;
+  }
+  long mask = (0xFFFFFFFFFFFFFFFFULL >> (64 - vlen));
```

Comment:
The higher bits of long input should be cleared. So we should generate an unsigned right shift instead of the signed one?

Reply:
I noticed that you used `ULL` suffix. So it should be fine. Please ignore above comment. Thanks!